File: backend/app/services/student_service.py
# backend/app/services/student_service.py
from app.models.student import Student
from app.utils.database import DatabaseConnection

class StudentService:
    def __init__(self):
        db = DatabaseConnection().get_database()
        self.student_model = Student(db)

    def add_student(self, student_data):
        return self.student_model.create_student(student_data)

    def get_student(self, student_id):
        return self.student_model.get_student_by_id(student_id)

    def update_student(self, student_id, update_data):
        return self.student_model.update_student(student_id, update_data)

    def delete_student(self, student_id):
        return self.student_model.delete_student(student_id)

    def list_students(self, skip=0, limit=100):
        return self.student_model.get_all_students(skip, limit)

    def get_students_by_class_stats(self):
        return self.student_model.get_student_count_by_class()

File: frontend/src/services/studentService.js
// frontend/src/services/studentService.js
import axios from 'axios';

// const API_URL = 'http://localhost:5000/api/students';
const API_URL = 'https://lis-portal.vercel.app/api/students';

export const createStudent = async (studentData) => {
    const response = await axios.post(API_URL, studentData);
    return response.data;
};

export const getStudents = async (skip = 0, limit = 100) => {
    const response = await axios.get(`${API_URL}?skip=${skip}&limit=${limit}`);
    return response.data;
};

export const getStudentById = async (studentId) => {
    const response = await axios.get(`${API_URL}/${studentId}`);
    return response.data;
};

export const updateStudent = async (studentId, studentData) => {
    const response = await axios.put(`${API_URL}/${studentId}`, studentData);
    return response.data;
};


export const deleteStudent = async (studentId) => {
    await axios.delete(`${API_URL}/${studentId}`);
};

